### PR TITLE
Fix Position decoding for negative coords

### PR DESCRIPTION
**Pull request by fenhl**
This fixes the `<BlockPos as Protocol>::proto_decode` method for positions with one or more negative coordinates.

It also removes some unnecessary parens from that module.

### CODE REVIEW

File: src/types/pos.rs [hunk 1]
```diff
@@ -31,16 +32,20 @@ impl Protocol for BlockPos {
...
 
     fn proto_decode(mut src: &mut Read) -> io::Result<[i32; 3]> {
         let block_pos = try!(src.read_u64::<BigEndian>());
         let x = (block_pos >> 38) as i32;
-        let y = ((block_pos >> 26) & 0xfff) as i32;
+        let y = (block_pos >> 26 & 0xfff) as i32;
         let z = (block_pos & 0x3ffffff) as i32;
-        Ok([x, y, z])
+        Ok([
+            if x >= 2.pow(25) { x - 2.pow(26) } else { x },
```
Comment (toqueteos) [inline src/types/pos.rs:45]: Why pow (which needs import) instead of module constants or just bit shift?

`2.pow(25)` == `1 << 25`

Comment (fenhl) [inline src/types/pos.rs:45]: I didn't think of that. Feel free to create a follow-up.

Comment (toqueteos) [inline src/types/pos.rs:45]: It's not a big change and I don't know if the compiler does not already will optimize that for us... Meh, I'll patch it later today, it's a minor thing.

Comment (fenhl) [inline src/types/pos.rs:45]: Done in #62, thanks @mindcat!

**Pull request merged**